src/lepus/utility/types/Transform.h

```diff
@@ -8,86 +8,180 @@ namespace lepus
+		m_Right.x(newRight.x());
+		m_Right.y(newRight.y());
+		m_Right.z(newRight.z());
+		m_Up.x(newUp.x());
+		m_Up.y(newUp.y());
+		m_Up.z(newUp.z());
+	    }
+
+	    static inline lepus::math::Matrix4x4 AxisAngle(lepus::types::Vector3& axis, float angle)
+	    {
+		lepus::math::Matrix4x4 ret = lepus::math::Matrix4x4::Identity();
+
+		lepus::types::Vector3 unitAxis = axis * (1.f / axis.Magnitude());
+
+		float c = cosf(angle);
+		float s = sinf(angle);
+
+		ret.set<0, 0>(c + (unitAxis.x() * unitAxis.x() * (1.f - c)));
+		ret.set<1, 0>((unitAxis.y() * unitAxis.x() * (1.f - c)) + (unitAxis.z() * s));
+		ret.set<2, 0>((unitAxis.z() * unitAxis.x() * (1.f - c)) - (unitAxis.y() * s));
+
+		ret.set<0, 1>((unitAxis.y() * unitAxis.x() * (1.f - c)) - (unitAxis.z() * s));
+		ret.set<1, 1>(c + (unitAxis.y() * unitAxis.y() * (1.f - c)));
+		ret.set<2, 1>((unitAxis.z() * unitAxis.y() * (1.f - c)) + (unitAxis.x() * s));
+
+		ret.set<0, 2>((unitAxis.z() * unitAxis.x() * (1.f - c)) + (unitAxis.y() * s));
+		ret.set<1, 2>((unitAxis.z() * unitAxis.y() * (1.f - c)) - (unitAxis.x() * s));
+		ret.set<2, 2>(c + (unitAxis.z() * unitAxis.z() * (1.f - c)));
+
+		return ret;
+	    }
+
+	    inline lepus::types::Vector3 Scale() const { return m_Scale; }
+
+	    inline void SetPosition(float x, float y, float z)
+	    {
+		m_Origin.x(x);
+		m_Origin.y(y);
+		m_Origin.z(z);
+	    }
+
+	    inline void SetScale(float scale)
+	    {
+		this->SetScale(scale, scale, scale);
+	    }
+
+	    inline void SetScale(float x, float y, float z)
+	    {
+		m_Scale.x(x);
+		m_Scale.y(y);
+		m_Scale.z(z);
+	    }
+
+	    inline lepus::math::Matrix4x4 BuildMatrix() const
+	    {
+		lepus::math::Matrix4x4 model = lepus::math::Matrix4x4::Identity();
+
+		// Translation
+		model.set<0, 3>(m_Origin.x());
+		model.set<1, 3>(m_Origin.y());
+		model.set<2, 3>(m_Origin.z());
+
+		// TODO: add scaling and rotation
+		lepus::types::Quaternion normRot = m_Rotation.Normalised();
+		lepus::types::Vector3 axis = normRot.Axis();
+		// axis = axis * (1.f / axis.Magnitude());
+		float angle = normRot.Angle();
+
+		float cosTheta = cosf(angle);
+		float invCosTheta = 1.f - cosTheta;
+		float sinTheta = sinf(angle);
+		model.set<0, 0>(m_Scale.x() * ((axis.x() * axis.x()) * invCosTheta + cosTheta));
+		model.set<0, 1>(m_Scale.y() * (axis.x() * axis.y() * invCosTheta + axis.z() * sinTheta));
+		model.set<1, 1>(m_Scale.y() * ((axis.y() * axis.y()) * invCosTheta + cosTheta));
+		model.set<1, 0>(m_Scale.x() * (axis.y() * axis.x() * invCosTheta - axis.z() * sinTheta));
+		model.set<2, 0>(m_Scale.x() * (axis.z() * axis.x() * invCosTheta + axis.y() * sinTheta));
+		model.set<2, 1>(m_Scale.y() * (axis.z() * axis.y() * invCosTheta - axis.x() * sinTheta));
+		model.set<0, 2>(m_Scale.z() * (axis.x() * axis.z() * invCosTheta - axis.y() * sinTheta));
+		model.set<1, 2>(m_Scale.z() * (axis.y() * axis.z() * invCosTheta + axis.x() * sinTheta));
+		model.set<2, 2>(m_Scale.z() * ((axis.z() * axis.z()) * invCosTheta + cosTheta));
```

Comment:
would be nice to order this matrix construction in either row-major order just so it's consistent to read